core/commands/dns.go

```diff
@@ -63,7 +63,7 @@ The resolver will give:
 			depth = namesys.DefaultDepthLimit
 		}
 		output, err := resolver.ResolveN(req.Context().Context, name, depth)
-		if err != nil {
+		if err != nil && (err != namesys.ErrResolveRecursion || depth > 1) {
```

Comment:
On Tue, Jun 09, 2015 at 04:05:21PM -0700, Juan Batiz-Benet wrote:
> > -		if err != nil {
> > +		if err != nil && (err != namesys.ErrResolveRecursion || depth > 1) {
> 
> what happens in this case, then, if it skipps the error return?
> what's the UX?

It's not an error in the depth=1 case, it just means we didn't bottom
out on the resolution in a single step.  Before this commit:

  $ ipfs resolve /ipns/tremily.us
  Error: could not resolve name (recursion limit exceeded).

After this commit:

  $ ipfs resolve /ipns/tremily.us
  /ipns/QmbqDJaoZYoGNw4dz7FqFDCf6q9EcHoMBQtoGViVFw2qv7

In both cases:

  $ ipfs resolve --recursive /ipns/tremily.us
  /ipfs/QmV2FrBtvue5ve7vxbAzKz3mTdWq8wfMNPwYd8d9KHksCF